word_language_model/main.py

```diff
@@ -130,14 +157,24 @@ def evaluate(data_source):
     model.eval()
     total_loss = 0.
     ntokens = len(corpus.dictionary)
-    hidden = model.init_hidden(eval_batch_size)
+    if args.model == 'Transformer':
+        tgt_mask = model.transformer.generate_square_subsequent_mask(args.bptt).to(device)
+    else:
+        hidden = model.init_hidden(eval_batch_size)
     with torch.no_grad():
         for i in range(0, data_source.size(0) - 1, args.bptt):
             data, targets = get_batch(data_source, i)
-            output, hidden = model(data, hidden)
+
+            if args.model == 'Transformer':
+                if args.bptt > len(data_source) - 1 - i:
+                    tgt_mask = model.transformer.generate_square_subsequent_mask(len(data_source) - 1 - i).to(device)
```

Comment:
Is there any way to do this implicitly within model and cache it via say `self.tgt_mask`?

Reply:
Should this be len(data) instead of len(data_source)?

Reply:
When we initiate the transformer model, we have no idea the length of tgt and src, unless we pass it through as an argument. The masks will change depending on the incoming tgt and src sequence. So it's better to generate them at the runtime.

Reply:
No. This is for the case when the data left don't have a full length of args.bptt.